### PR TITLE
fix: add "account" subsection to mobile web nav

### PR DESCRIPTION
**Pull request by adamvert**
https://artsyproduct.atlassian.net/browse/FX-1848

image
image

The Inbox link appears between _Works for you_ and _Account_, i.e. in the top level, not the subnav - I hope that's correct?

### CODE REVIEW

File: src/v2/Components/NavBar/Menus/MobileNavMenu/MobileLink.tsx [hunk 1]
```diff
@@ -47,7 +48,7 @@ export const MobileLink: React.FC<MobileLinkProps> = ({
           <Link
             href={href}
             underlineBehavior="none"
-            onClick={() => handleClickTracking(href)}
+            onClick={onClick || (() => handleClickTracking(href))}
```
Comment (dzucconi) [inline src/v2/Components/NavBar/Menus/MobileNavMenu/MobileLink.tsx:51]: Should probably wrap this rather than bypass the click tracking?

File: src/v2/Components/NavBar/Menus/MobileNavMenu/MobileNavMenu.tsx [hunk 2]
```diff
@@ -291,7 +295,7 @@ const AuthenticateLinks: React.FC = () => {
     <Box>
       <Separator my={1} color={color("black10")} />
       <MobileLink href={authLink(ModalType.signup)}>Sign Up</MobileLink>
-      <MobileLink href={authLink(ModalType.login)}>Login</MobileLink>
+      <MobileLink href={authLink(ModalType.login)}>Log In</MobileLink>
```
Comment (dzucconi) [inline src/v2/Components/NavBar/Menus/MobileNavMenu/MobileNavMenu.tsx:298]: The casing should be consistent with desktop?

Comment (adamvert) [reply src/v2/Components/NavBar/Menus/MobileNavMenu/MobileNavMenu.tsx:298]: Right - I took the case from `Sign Up` in the line above, but I'll change them both to match desktop.

File: src/v2/Components/NavBar/Menus/MobileNavMenu/__tests__/MobileNavMenu.jest.tsx [hunk 3]
```diff
@@ -31,6 +30,13 @@ describe("MobileNavMenu", () => {
     )
   }
 
+  const getMobileMenuLinkContainer = (userType = null, lab_features = []) =>
```
Comment (xtina-starr) [inline src/v2/Components/NavBar/Menus/MobileNavMenu/__tests__/MobileNavMenu.jest.tsx:33]: Nice!!

File: .vscode/launch.json [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  "configurations": [
+    {
+      "type": "pwa-chrome",
+      "request": "launch",
+      "name": "Launch Chrome against localhost",
+      "url": "http://localhost:8080",
+      "webRoot": "${workspaceFolder}"
+    }
+  ]
+}
```
Comment (xtina-starr) [inline .vscode/launch.json:15]: Was this file intended to be checked in?

Comment (anandaroop) [reply .vscode/launch.json:15]: I wondered as well — did also notice that we have this VS Code debugger helper [in various projects](https://github.com/search?q=org%3Aartsy+path%3A.vscode&type=Code)

File: src/v2/Components/NavBar/Menus/MobileNavMenu/LoggedInLinks.tsx [hunk 5]
```diff
@@ -16,6 +17,39 @@ export const LoggedInLinks: React.FC<
...
+      {
+        text: "Log out",
+        href: "#logout",
+        onClick: event => {
+          event.preventDefault()
+          mediator.trigger("auth:logout")
+        },
+      },
+    ],
+  }
```
Comment (dzucconi) [inline src/v2/Components/NavBar/Menus/MobileNavMenu/LoggedInLinks.tsx:52]: Why move this out of JSX?

File: .vscode/launch.json (hunk 4 above)
Comment (damassi) [reply .vscode/launch.json:15]: Iif there's a launch config that's helpful to the wider team we should announce!

File: src/v2/Components/NavBar/Menus/MobileNavMenu/MobileNavMenu.tsx [hunk 6]
```diff
@@ -261,29 +260,50 @@ const AuthenticateLinks: React.FC = () => {
   )
 }
 
-export const LoggedInLinks: React.FC<any> = ({ account }) => {
+export const LoggedInLinks: React.FC<any> = () => {
   const { mediator, user } = useSystemContext()
+  let menu = {
+    title: "Account",
+    links: [
```
[the remark this answers is not in the document]
Comment (adamvert) [reply src/v2/Components/NavBar/Menus/MobileNavMenu/MobileNavMenu.tsx]: Exactly: since the desktop implementation is already bypassing the JSON definition (and has slightly different structure), it seemed like a pointless complication to put the mobile entries there

File: src/v2/Components/NavBar/Menus/MobileNavMenu/LoggedInLinks.tsx (hunk 5 above)
Comment (adamvert) [reply src/v2/Components/NavBar/Menus/MobileNavMenu/LoggedInLinks.tsx:52]: `<MobileSubmenuLink>` is already set up to work with this kind of structure, but it seemed cleaner and more readable to put it in a `const` rather than embedding it directly into the JSX. I tend to aim for clean JSX with references to pre-defined vars as much as possible, cos I find that JSX is already obfuscated enough without filling it up with JS structures. I'm very happy to have my mind changed on this though, if you think embedding is more readable?

File: .vscode/launch.json (hunk 4 above)
Comment (adamvert) [reply .vscode/launch.json:15]: Yeah, actually this was a mistake 😅 - I spent a while configuring things so I could debug the tests within VSCode, but I didn't end up using this Chrome profile anyway, so I'll just quietly remove it...

**Comment by adamvert** [comment]
> I added the `unreadNotificationsCount` to the query when I created the `LoggedInLinks` component but turns out it isn't needed (we only use `unreadConversationCount`). We could remove it from the query in this PR.
> 
> Seeing as though it's out of the scope for the work you were doing I'm also happy to address it in a follow PR but I figured I'd mention it.

Hey @xtina-starr! I can do it here if you like, but I generally prefer to keep PRs atomic and not hide off-topic commits within them.

File: src/v2/Components/NavBar/Menus/MobileNavMenu/LoggedInLinks.tsx (hunk 5 above)
Comment (dzucconi) [reply src/v2/Components/NavBar/Menus/MobileNavMenu/LoggedInLinks.tsx:52]: Ah I see 👍 — though I would pull it out into a real const outside of the closure here.

Comment (adamvert) [reply src/v2/Components/NavBar/Menus/MobileNavMenu/LoggedInLinks.tsx:52]: Yeah, I thought that too, but I need access to the mediator to trigger a logout event, and that comes from the `useSystemContext` hook

Comment (dzucconi) [reply src/v2/Components/NavBar/Menus/MobileNavMenu/LoggedInLinks.tsx:52]: 👍 👍

**Comment by anandaroop** [comment]
Nice, gonna give this a merge, and let's address the notifications count change in a follow-up 🎉
